pennylane/transforms/get_dag_commutation.py

```diff
@@ -0,0 +1,1810 @@
+    you can directly access all node attributes.
+
+    >>> second_node = dag.get_node(2)
+    >>> second_operation = second_node.op
+    >>> second_node_successors = second_node.successors
+    >>> second_node_predecessors = second_node.predecessors
+
+
+    For more details, see:
+
+    * Iten, R., Moyard, R., Metger, T., Sutter, D., Woerner, S.
+      "Exact and practical pattern matching for quantum circuit optimization" `arXiv:1909.05270. (2020).
+      <https://arxiv.org/pdf/1909.05270.pdf>`_
+
+    """
+
+    # pylint: disable=protected-access
+
+    @wraps(circuit)
+    def wrapper(*args, **kwargs):
+
+        if isinstance(circuit, qml.QNode):
+            # user passed a QNode, get the tape
+            circuit.construct(args, kwargs)
+            tape = circuit.qtape
+
+        elif isinstance(circuit, qml.tape.QuantumTape):
+            # user passed a tape
+            tape = circuit
+
+        elif callable(circuit):
+            # user passed something that is callable but not a tape or qnode.
+            tape = qml.transforms.make_tape(circuit)(*args, **kwargs)
+            # raise exception if it is not a quantum function
+            if len(tape.operations) == 0:
+                raise ValueError("Function contains no quantum operation")
+
+        else:
+            raise ValueError("Input is not a tape, QNode, or quantum function")
+
+        # if no wire ordering is specified, take wire list from tape
+        wires = tape.wires
+
+        consecutive_wires = Wires(range(len(wires)))
+        wires_map = OrderedDict(zip(wires, consecutive_wires))
+
+        for obs in tape.observables:
+            obs._wires = Wires([wires_map[wire] for wire in obs.wires.tolist()])
+
+        with qml.tape.Unwrap(tape):
+            # Initialize DAG
+            dag = CommutationDAG(consecutive_wires, tape.observables)
+
+            for operation in tape.operations:
+                operation._wires = Wires([wires_map[wire] for wire in operation.wires.tolist()])
+                dag.add_node(operation)
+            dag._add_successors()
+        return dag
+
+    return wrapper
+
+
+position = OrderedDict(
+    {
+        "Hadamard": 0,
+        "PauliX": 1,
+        "PauliY": 2,
+        "PauliZ": 3,
+        "SWAP": 4,
+        "ctrl": 5,
+        "S": 6,
+        "T": 7,
+        "SX": 8,
+        "ISWAP": 9,
+        "SISWAP": 10,
+        "Barrier": 11,
+        "WireCut": 12,
+        "RX": 13,
+        "RY": 14,
+        "RZ": 15,
+        "PhaseShift": 16,
+        "Rot": 17,
+        "MultiRZ": 18,
+        "Identity": 19,
+        "U1": 20,
+        "U2": 21,
+        "U3": 22,
+        "IsingXX": 23,
+        "IsingYY": 24,
+        "IsingZZ": 25,
+        "QubitStateVector": 26,
+        "BasisState": 27,
+    }
+)
+"""OrderedDict[str, int]: represents the place of each gates in the commutation_map."""
+
+commutation_map = OrderedDict(
+    {
```

Comment:
You can use

```python
# fmt: off
commutation_map = OrderedDict(
    ...
)
# fmt: on
```

to make black ignore this object :) I think that would make a lot of sense, since the way black is currently formatting it is a bit unwieldy.

Reply:
Thank you, I didn't know this trick!